devU-client/src/components/pages/containerAutoGraderForm.tsx

```diff
@@ -0,0 +1,82 @@
+import React,{useState} from 'react'
+import PageWrapper from 'components/shared/layouts/pageWrapper'
+import styles from './nonContainerAutoGraderForm.scss'
+import TextField from 'components/shared/inputs/textField'
+import Button from 'components/shared/inputs/button'
+import { useActionless } from 'redux/hooks'
+import { SET_ALERT } from 'redux/types/active.types'
+import RequestService from 'services/request.service'
+
+const ContainerAutoGraderForm = () => {
+    const [setAlert] = useActionless(SET_ALERT)
+
+    const [graderFile, setGraderFile] = useState<File | null>()
+    const [makefile, setMakefile] = useState<File | null>()
+    const [formData,setFormData] = useState({
+        assignmentId: '',
+        autogradingImage: '',
+        timeout: '',
+    })
+
+    const handleChange = (value: String, e : React.ChangeEvent<HTMLInputElement>) => {
+        const key = e.target.id
+        setFormData(prevState => ({...prevState,[key] : value}))
+    }
+    //This is janky but it works and I'm too tired to come up with a better solution
+    //this is done because the files need to be uniquely identified for multer to parse them from the multipart
+    const handleGraderfileChange = (e : React.ChangeEvent<HTMLInputElement>) => {
+        setGraderFile(e.target.files?.item(0))
+    }
+    const handleMakefileChange = (e : React.ChangeEvent<HTMLInputElement>) => {
+        setMakefile(e.target.files?.item(0))
+    }
+
+    const handleSubmit = () => {
+
+        const multipart = new FormData
+        multipart.append('assignmentId', formData.assignmentId)
+        multipart.append('autogradingImage', formData.autogradingImage)
+        multipart.append('timeout', String(formData.timeout))
+        if (graderFile) multipart.append('graderFile', graderFile)
```

Comment:
graderFile should be required but makefile is not

Reply:
That if statement is only there due to some typescript annoyances, if there isn't a graderfile it should be caught by the validator